### PR TITLE
to-hex payloadprocessor

### PR DESCRIPTION
**Pull request by EN10**
Useful script I used for Natas 19 Challenge:
https://youtu.be/aZ3ZfAhdqT8

### CODE REVIEW

**Comment by psiinon** [comment]
Thanks for sharing this :)
Could you add some details in the script, maybe including a link to that video?
Once its merged people wont see this commit so a bit of context would be useful.

**Comment by EN10** [comment]
> Thanks for sharing this :)
> Could you add some details in the script, maybe including a link to that video?
> Once its merged people wont see this commit so a bit of context would be useful.

thanks, this is my first ever pull :)

**Comment by EN10** [comment]
I think I'm done with edits?

**Comment by psiinon** [comment]
Looks good to me :)
Just one more thing, could you add a comment to the changelog: https://github.com/zaproxy/community-scripts/blob/master/CHANGELOG.md

File: payloadprocessor/to_hex.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+ /**
+ * Converts a string payload to hex.
+ * 
+ * Created to add functinality found in Burp to solve Natas19
```
Comment (thc202) [inline payloadprocessor/to_hex.js]: There's a typo in functionality.

File: payloadprocessor/to_hex.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * https://www.youtube.com/watch?v=z3RtpWZ_R3Q
+ *
+ */
+
+function toHex(payload) {
```
Comment (thc202) [inline payloadprocessor/to_hex.js]: It needs to keep the previous name to work properly.

File: payloadprocessor/to_hex.js (hunk 1 above)
Comment (EN10) [reply payloadprocessor/to_hex.js]: oops

File: payloadprocessor/to_hex.js (hunk 2 above)
Comment (EN10) [reply payloadprocessor/to_hex.js]: ok

**Comment by EN10** [comment]
fixed typo, function name and added to changelog

**Comment by EN10** [comment]
is there anything else required?

**Comment by thc202** [comment]
Could you squash the commits? (The first is using your email, fine if you want it public though.)

**Comment by EN10** [comment]
> Could you squash the commits? (The first is using your email, fine if you want it public though.)

how do I do that?

**Comment by kingthorin** [comment]
Here you go:

- https://www.instructables.com/id/How-to-Rebase-Interactive-With-Eclipse-eGit/
- https://www.kernel.org/pub/software/scm/git/docs/git-rebase.html#_interactive_mode
- https://wiki.eclipse.org/EGit/User_Guide#Interactive_Rebase

**Comment by EN10** [comment]
> Here you go:
> 
> * https://www.instructables.com/id/How-to-Rebase-Interactive-With-Eclipse-eGit/
> * https://www.kernel.org/pub/software/scm/git/docs/git-rebase.html#_interactive_mode
> * https://wiki.eclipse.org/EGit/User_Guide#Interactive_Rebase

ok thanks, I'm using Bash rather than Eclipse.
So I used: git rebase --autostash
hope that works?

**Comment by kingthorin** [comment]
The middle link has CLI details. These might also help you:
- https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history
- https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d

**Comment by kingthorin** [comment]
If you're having trouble, let me know I can tackle it.

**Comment by thc202** [comment]
Some more resources:
 - https://help.github.com/en/github/using-git/setting-your-username-in-git
 - https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address
 - https://help.github.com/en/github/using-git/about-git-rebase

**Comment by EN10** [comment]
> If you're having trouble, let me know I can tackle it.

thanks, for the help.

i'm getting this issue:
```
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_ro_RO/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_ru_RU/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_si_LK/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_sk_SK/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_sl_SI/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_sq_AL/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_sr_CS/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_sr_SP/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_tr_TR/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_ur_PK/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_zh_CN/contents/images/script-share.png: needs update
You must edit all merge conflicts and then
mark them as resolved using git add
```
seems to be stopping me from rebase

**Comment by thc202** [comment]
You need to force push (`-f` flag).

**Comment by EN10** [comment]
> You need to force push (`-f` flag).

thanks for your patience, i've realised i have edited a bunch of images in my last commit and trying to undo that.

**Comment by kingthorin** [comment]
Check `git status` to see what it thinks is pending.

**Comment by EN10** [comment]
> Check `git status` to see what it thinks is pending.

git status

    On branch master
    Your branch is up-to-date with 'origin/master'.
    Changes not staged for commit:
      (use "git add <file>..." to update what will be committed)
      (use "git checkout -- <file>..." to discard changes in working directory)
    modified:   src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_tr_TR/contents/images/script-share.png
    modified:   src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_ur_PK/contents/images/script-share.png
    modified:   src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_zh_CN/contents/images/script-share.png

**Comment by EN10** [comment]
ok i think i've squashed the commits, and removed email.

**Comment by thc202** [comment]
The committer name/email is probably not correct.

**Comment by EN10** [comment]
> The committer name/email is probably not correct.

thanks, I used this to fix for reference.

https://stackoverflow.com/questions/3042437/how-to-change-the-commit-author-for-one-specific-commit

**Comment by thc202** [comment]
Thank you!

**Pull request merged**